src/cross_toml.rs

```diff
@@ -227,11 +332,225 @@ mod tests {
             build-std = true
             image = "test-image"
         "#;
-        let (parsed_cfg, unused) = CrossToml::parse(test_str)?;
+        let (parsed_cfg, unused) = CrossToml::parse_from_cross(test_str)?;
 
         assert_eq!(parsed_cfg, cfg);
         assert!(unused.is_empty());
 
         Ok(())
     }
+
+    #[test]
+    pub fn parse_from_empty_cargo_toml() -> Result<()> {
+        let test_str = r#"
+          [package]
+          name = "cargo_toml_test_package"
+          version = "0.1.0"
+
+          [dependencies]
+          cross = "1.2.3"
+        "#;
+
+        let res = CrossToml::parse_from_cargo(test_str)?;
+        assert!(res.is_none());
+
+        Ok(())
+    }
+
+    #[test]
+    pub fn parse_from_cargo_toml() -> Result<()> {
+        let cfg = CrossToml {
+            targets: HashMap::new(),
+            build: CrossBuildConfig {
+                env: CrossEnvConfig {
+                    passthrough: vec![],
+                    volumes: vec![],
+                },
+                build_std: None,
+                xargo: Some(true),
+                default_target: None,
+            },
+        };
+
+        let test_str = r#"
+          [package]
+          name = "cargo_toml_test_package"
+          version = "0.1.0"
+
+          [dependencies]
+          cross = "1.2.3"
+
+          [package.metadata.cross.build]
+          xargo = true
+        "#;
+
+        if let Some((parsed_cfg, _unused)) = CrossToml::parse_from_cargo(test_str)? {
+            assert_eq!(parsed_cfg, cfg);
+        } else {
+            panic!("Parsing result is None");
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    pub fn merge() -> Result<()> {
```

Comment:
Tests are great, but this is a bit too much in my opinion.

I think the better approach is to parse the structs as strings, that way we skip having to update these whenever a field is added to the config structs and it's smaller in length.